src/symbol_table.rs

```diff
@@ -1,14 +1,119 @@
+use std::borrow::Borrow;
 use std::collections::HashMap;
+use std::hash::{Hash, Hasher};
+use std::ops::Deref;
+use std::rc::Rc;
 
 use crate::constants::v1_0;
 use crate::types::SymbolId;
 
+/// Stores or points to the text of a given [Symbol].
+#[derive(Debug, Eq)]
+enum SymbolText {
+    // This Symbol refers to a string in the symbol table
```

Comment:
Should we specify here whether it points to local symbol table or shared symbol tables or both? thoughts?

Reply:
A recent clarification to the Ion spec says that at the application level, symbols must be represented as text. If a symbol ID is encountered that cannot be resolved to text, the reader can raise an error. (Readers can _try_ to keep processing the data without the relevant text if the task at hand doesn't require it, but are not required to do so by the spec. That's an API decision.)

In light of this distinction, I think we should either:
* Ask folks who are interested in surfacing lower-level encoding artifacts like symbol tables and IDs to use a different interface (like `SystemReader`, maybe?)
  AND/OR
* Wait until someone has a use case for storing provenance in the `Symbol` itself

In either case, I'd like to hold off on trying to add that now.